components/html-block/html-block.js

```diff
@@ -226,42 +229,54 @@ class HtmlBlock extends LitElement {
 		`;
 	}
 
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._templateObserver) this._templateObserver.disconnect();
+	}
+
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
-		if (!this._renderContainer) {
+		if (this._renderContainer) return;
 
-			this.shadowRoot.innerHTML = '<div class="d2l-html-block-rendered"></div><slot></slot>';
+		this.shadowRoot.innerHTML = '<div class="d2l-html-block-rendered"></div><slot></slot>';
 
-			this.shadowRoot.querySelector('slot').addEventListener('slotchange', async e => {
```

Comment:
This is likely the most interesting part of these changes and probably easier viewed split rather than unified.  The stamping logic has just been factored out so that it can be called for the two cases : 1) when the slotted template element is initially added or replaced; and 2) when the content of the template changes.

Case (2) is expected to be fairly common because when Lit consumers render this component, they will not be replacing their initially rendered `template` element, but rather the contents of it when they want to update.  This is because Lit will not recreate that `template` element.  To accommodate this scenario, a MutationObserver is wired up for the `template.content` to re-stamp the content.